tests/utils/urls.py

```diff
@@ -122,12 +122,16 @@ def assert_valid_url(url, location=None, status_code=requests.codes.moved_perman
             # strip off query for further comparison
             resp_location = resp_location.split('?')[0]
 
-        assert location == unquote(resp_location).decode()
+        assert location == unquote(resp_location)
 
     if resp_headers and not follow_redirects:
+
+        def convert_to_set(header):
+            return frozenset(d.strip() for d in header.lower().split(','))
```

Comment:
Not wrong but why `frozenset` and not regular `set`?

Reply:
For this case, the set will never be modified.